File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.js
function solution(s) {
    let answer = 0;
    
    
    for (let i = 0; i < s.length; i++) {
        let stack = [];
        
        let newS = s.substr(i);
        newS += s.substr(0, i);
        
        let flag = true;
        for (let j = 0; j < newS.length; j++) {
            if (newS[j] === '(' || newS[j] === '{' || newS[j] === '[') {
                stack.push(newS[j]);
            }
            else {
                if (newS[j] === ')' && stack[stack.length-1] === '(') {
                    stack.pop();
                }
                else if (newS[j] === '}' && stack[stack.length-1] === '{') {
                    stack.pop();
                }
                else if (newS[j] === ']' && stack[stack.length-1] === '[') {
                    stack.pop();
                }
                else {
                    flag = false;
                    break;
                }
            }
        }
        
        if (stack.length === 0 && flag) {
            answer++;
        }
    }
    
    
    return answer;
}

File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.py
from collections import deque

def check(deq):
    
    stack = []
    
    for elem in deq:
        if elem == ']':
            if stack and stack[-1] == '[':
                stack.pop()
            else:
                return False
        elif elem == '}':
            if stack and stack[-1] == '{':
                stack.pop()
            else:
                return False
        elif elem == ')':
            if stack and stack[-1] == '(':
                stack.pop()
            else:
                return False
        else:
            stack.append(elem)
            
    if stack:
        return False
    else:
        return True
    

def solution(s):
    answer = 0
    
    deq = deque()
    
    for i in range(len(s)):
        deq.append(s[i])
        
    
    if check(deq):
            answer += 1
    for i in range(len(deq)-1):
        deq.append(deq.popleft())
        if check(deq):
            answer += 1
    
    return answer